### PR TITLE
improve pipeline context implementation

### PR DESCRIPTION
**Pull request by srdjan-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/common/components/PipelineStudio/PipelineLoaderContext/PipelineLoaderContext.tsx [hunk 1]
```diff
@@ -74,98 +69,75 @@ export function PipelineLoaderProvider({
...
+  React.useEffect(() => {
+    if (typeof routerState?.yamlSyntax !== 'undefined') {
+      setYamlVersion(routerState?.yamlSyntax)
+    }
+  }, [routerState?.yamlSyntax])
+
   const abortControllerRef = React.useRef<AbortController | null>(null)
   React.useEffect(() => {
-    // fetch pipeline after trying to initialize idb
+    // NOTE: have to wait for both api od idb because we are using idb in child contexts
```
Comment (rameshx) [inline src/modules/70-pipeline/common/components/PipelineStudio/PipelineLoaderContext/PipelineLoaderContext.tsx:122]: looks like there's a typo

File: src/modules/70-pipeline/common/components/PipelineStudio/IDBContext/utils.ts [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
   gitDetails,
-  idbDelete
+  idb
 }: {
   routeState: RouteState
   identifier: string
   gitDetails?: EntityGitDetails
-  idbDelete: IDBContextInterface['del']
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  idb: IDB<any>
```
Comment (rameshx) [inline src/modules/70-pipeline/common/components/PipelineStudio/IDBContext/utils.ts:16]: Is it possible to give the correct type here?

File: src/modules/70-pipeline/components/PipelineStudio/PipelineContext/PipelineAsyncActions.ts [hunk 3]
```diff
@@ -180,19 +174,17 @@ export const processPipelineFromAPIAction = (
...
   return async (dispatch, getState) => {
-    const { idbGet, idbPut } = props
-
-    const { supportingTemplatesGitx } = props
+    const { idb, initialLoading, supportingTemplatesGitx } = props
     const { routeState, storeMetadata } = getState()
     const {
       //forceFetch = false,
-      forceUpdate = false,
+      //forceUpdate = false,
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineContext/PipelineAsyncActions.ts:187]: Can these be removed?

File: src/modules/70-pipeline/common/components/PipelineStudio/PipelineLoaderContext/PipelineLoaderContext.tsx (hunk 1 above)
Comment (srdjan-harness) [reply src/modules/70-pipeline/common/components/PipelineStudio/PipelineLoaderContext/PipelineLoaderContext.tsx:122]: remove comment

File: src/modules/70-pipeline/common/components/PipelineStudio/IDBContext/utils.ts (hunk 2 above)
Comment (srdjan-harness) [reply src/modules/70-pipeline/common/components/PipelineStudio/IDBContext/utils.ts:16]: plan to use from template module. replaced with  generic.

**Review by rameshx — APPROVED** [review]

**Pull request merged**